File: mobile/ios/AdMobConsentHelper.swift
import UIKit
import UserMessagingPlatform

/// Google UMP — call before GADMobileAds.sharedInstance().start(...)
/// @see docs/admob-ump-native.md
enum AdMobConsentHelper {

  static func requestConsent(from viewController: UIViewController, completion: @escaping () -> Void) {
    let params = RequestParameters()
    ConsentInformation.shared.requestConsentInfoUpdate(with: params) { _ in
      ConsentForm.loadAndPresentIfRequired(from: viewController) { _ in
        if ConsentInformation.shared.canRequestAds {
          completion()
        }
      }
    }
  }

  /// Settings → Privacy and cookie settings (AdMob revocation link requirement).
  static func showPrivacyOptions(from viewController: UIViewController) {
    ConsentForm.presentPrivacyOptionsForm(from: viewController) { _ in }
  }
}

File: mobile/android/AdMobConsentHelper.kt
package com.shirwell.music

import android.app.Activity
import com.google.android.ump.ConsentRequestParameters
import com.google.android.ump.UserMessagingPlatform

/**
 * Google UMP — run before MobileAds.initialize.
 * @see docs/admob-ump-native.md
 */
object AdMobConsentHelper {

  fun requestConsent(activity: Activity, onReady: () -> Unit) {
    val params = ConsentRequestParameters.Builder().build()
    val consentInfo = UserMessagingPlatform.getConsentInformation(activity)
    consentInfo.requestConsentInfoUpdate(
      activity,
      params,
      {
        UserMessagingPlatform.loadAndShowConsentFormIfRequired(activity) { _ ->
          if (consentInfo.canRequestAds()) onReady()
        }
      },
      { onReady() },
    )
  }

  /** Settings → Privacy and cookie settings (AdMob revocation link requirement). */
  fun showPrivacyOptions(activity: Activity, onDismiss: () -> Unit = {}) {
    UserMessagingPlatform.showPrivacyOptionsForm(activity) { onDismiss() }
  }
}
